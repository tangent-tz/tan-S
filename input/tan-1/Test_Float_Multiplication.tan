main {
    # Test 1: multiplying two positive floating-point numbers
    var floatone := 3.14;
    var floattwo := 2.5;
    var resultone := floatone * floattwo;
    print resultone;
    print \n;

    # Test 2: multiplying a positive floating-point number with a negative floating-point number
    floatone := 3.14;
    floattwo := -2.5;
    var resulttwo := floatone * floattwo;
    print resulttwo;
    print \n;

    # Test 3: multiplying a negative floating-point number with two negative floating-point number
    floatone := -3.14;
    floattwo := -2.5;
    var resultthree := floatone * floattwo;
    print resultthree;
    print \n;

    # Test 4: multiplying a floating-point number by zero
    floatone := 3.14;
    floattwo := 0.0;
    var resultfour := floatone * floattwo;
    print resultfour;
    print \n;

    # Test 5: multiplying floating-point numbers with different decimal precision
    floatone := 3.14159;
    floattwo := 2.71828;
    var resultfive := floatone * floattwo;
    print resultfive;
    print \n;

    # Test 6: test the associativity of the multiplication operation
    floatone := 2.0;
    floattwo := 3.0;
    var floatthree := 4.0;
    var resultsix := floatone * floattwo * floatthree;
    print resultsix;
    print \n;

    # Test 7: test for any possible edge cases or corner cases
    # edge case: multiplying zero by any floating-point number
    floatone := 0.0;
    floattwo := 3.14;
    var resultseven := floatone * floattwo;
    print resultseven;
    print \n;
}
