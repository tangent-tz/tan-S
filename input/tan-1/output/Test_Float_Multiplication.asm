        Jump         $$main                    
        DLabel       $eat-location-zero        
        DataZ        8                         
        DLabel       $print-format-integer     
        DataC        37                        %% "%d"
        DataC        100                       
        DataC        0                         
        DLabel       $print-format-boolean     
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-float       
        DataC        37                        %% "%f"
        DataC        102                       
        DataC        0                         
        DLabel       $print-format-character   
        DataC        37                        %% "%c"
        DataC        99                        
        DataC        0                         
        DLabel       $print-format-string      
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-newline     
        DataC        10                        %% "\n"
        DataC        0                         
        DLabel       $print-format-space       
        DataC        32                        %% " "
        DataC        0                         
        DLabel       $print-format-tab         
        DataC        9                         %% "\t"
        DataC        0                         
        DLabel       $boolean-true-string      
        DataC        116                       %% "true"
        DataC        114                       
        DataC        117                       
        DataC        101                       
        DataC        0                         
        DLabel       $boolean-false-string     
        DataC        102                       %% "false"
        DataC        97                        
        DataC        108                       
        DataC        115                       
        DataC        101                       
        DataC        0                         
        DLabel       $errors-general-message   
        DataC        82                        %% "Runtime error: %s\n"
        DataC        117                       
        DataC        110                       
        DataC        116                       
        DataC        105                       
        DataC        109                       
        DataC        101                       
        DataC        32                        
        DataC        101                       
        DataC        114                       
        DataC        114                       
        DataC        111                       
        DataC        114                       
        DataC        58                        
        DataC        32                        
        DataC        37                        
        DataC        115                       
        DataC        10                        
        DataC        0                         
        Label        $$general-runtime-error   
        PushD        $errors-general-message   
        Printf                                 
        Halt                                   
        DLabel       $errors-int-divide-by-zero 
        DataC        105                       %% "integer divide by zero"
        DataC        110                       
        DataC        116                       
        DataC        101                       
        DataC        103                       
        DataC        101                       
        DataC        114                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$i-divide-by-zero        
        PushD        $errors-int-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $errors-float-divide-by-zero 
        DataC        102                       %% "float divide by zero"
        DataC        108                       
        DataC        111                       
        DataC        97                        
        DataC        116                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$f-divide-by-zero        
        PushD        $errors-float-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $usable-memory-start      
        DLabel       $global-memory-block      
        DataZ        80                        
        Label        $$main                    
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        3.140000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        2.500000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        16                        
        Add                                    %% resultone
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        16                        
        Add                                    %% resultone
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        3.140000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        2.500000                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        24                        
        Add                                    %% resulttwo
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        24                        
        Add                                    %% resulttwo
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        3.140000                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        2.500000                  
        FNegate      null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        32                        
        Add                                    %% resultthree
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        32                        
        Add                                    %% resultthree
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        3.140000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        0.000000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        40                        
        Add                                    %% resultfour
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        40                        
        Add                                    %% resultfour
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        3.141590                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        2.718280                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        48                        
        Add                                    %% resultfive
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        48                        
        Add                                    %% resultfive
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        2.000000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        3.000000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        56                        
        Add                                    %% floatthree
        PushF        4.000000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        64                        
        Add                                    %% resultsix
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        PushD        $global-memory-block      
        PushI        56                        
        Add                                    %% floatthree
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        64                        
        Add                                    %% resultsix
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        PushF        0.000000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        PushF        3.140000                  
        StoreF                                 
        PushD        $global-memory-block      
        PushI        72                        
        Add                                    %% resultseven
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% floatone
        LoadF                                  
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% floattwo
        LoadF                                  
        FMultiply    null                      
        StoreF                                 
        PushD        $global-memory-block      
        PushI        72                        
        Add                                    %% resultseven
        LoadF                                  
        PushD        $print-format-float       
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        Halt                                   
