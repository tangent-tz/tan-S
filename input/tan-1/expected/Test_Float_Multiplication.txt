Multiplying two positive floating-point numbers                                         : 7.85
Multiplying a positive floating-point number with a negative floating-point number      : -7.85
Multiplying a negative floating-point number with two negative floating-point number    : 7.85
Multiplying a floating-point number by zero                                             : 0.0
Multiplying floating-point numbers with different decimal precision                     : 8.5397
Test the associativity of the multiplication operation                                  : 24.0
Test for any possible edge cases or corner cases                                        : 0.0
